opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -1650,9 +1655,18 @@ namespace Opm
         // fetch potentials (sign is updated on the outside).
         well_potentials.clear();
         well_potentials.resize(np, 0.0);
-        for (int compIdx = 0; compIdx < this->num_components_; ++compIdx) {
-            const EvalWell rate = well_copy.primary_variables_.getQs(compIdx);
-            well_potentials[this->modelCompIdxToFlowCompIdx(compIdx)] = rate.value();
+        for (int phase = 0; phase < np; ++phase) {
```

Comment:
By definition, we use `comp_idx` for `getQs`. Do you have some concrete observation for the change in this part?

Reply:
Yes, that's what I though, but this is where the inclusion of solvent fails as `num_compenents_` is 4 and there's no `getQs` for that.

Reply:
The function `modelCompIdxToFlowCompIdx` is calling for component indices. 

I think we can do this, 
```c++
        for (int comp_idx = 0; comp_idx < this->num_components_; ++comp_idx) {
            if (has_solvent && comp_idx == Indices::contiSolventEqIdx) continue; // we do not store the solvent in the well_potentials
            const EvalWell rate = well_copy.primary_variables_.getQs(comp_idx);
            well_potentials[this->modelCompIdxToFlowCompIdx(comp_idx)] = rate.value();
        }
```

But it also reveals that potentially in multiple occasions, we might do things related to solvent not totally correct.

Reply:
Furthermore, if you want, you can incorporate the code below inside the if condition, `if (has_solvent && comp_idx == Indices::contiSolventEqIdx)` to add the solvent to the gas `well_potentials`.  We are basically merging two code blocks to be one and might benefit the future way of coding related to solvent.

Reply:
Thanks, your suggestion makes more sense. Have updated the PR.

Reply:
> Furthermore, if you want, you can incorporate the code below inside the if condition, `if (has_solvent && comp_idx == Indices::contiSolventEqIdx)` to add the solvent to the gas `well_potentials`. We are basically merging two code blocks to be one and might benefit the future way of coding related to solvent.

Ah, updated before I saw your comment